mixlib-shellout.gemspec

```diff
@@ -12,7 +16,7 @@ Gem::Specification.new do |s|
   s.homepage = "https://github.com/chef/mixlib-shellout"
   s.license = "Apache-2.0"
 
-  s.required_ruby_version = ">= 3.0"
+  s.required_ruby_version = "~> 3.4.1"
```

Comment:
This is a bad practice for multiple reasons:
1) Only ruby versions matching 3.4.x can use this version.
2) There probably isn't any incompatibility with 3.1 - 3.3 (3.0 may be another story)... I wouldn't be as aggressive with the minimum requirements... If it works on 3.1 and 3.2, then those, in my opinion, should be the floor.

Reply:
@tpowell-progress yeah I had `>=3.1` in https://github.com/chef/mixlib-shellout/commit/ac2617ff0fe712a8ddcf0f2563c79689fd12dbd4 but I'm forgetting why I changed it to this. Will revert it to `>=3.1`